src/main/java/Duke.java

```diff
@@ -1,10 +1,258 @@
+
+
+    public static void printLogoAndGreet() {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println(OUTPUT_DIVIDER);
+        System.out.println(MESSAGE_GREET_USER);
+        System.out.println(OUTPUT_DIVIDER);
+    }
+
+    public static void printTaskList(Task[] tasks) {
+        System.out.println(OUTPUT_DIVIDER);
+        System.out.println(MESSAGE_LIST_TASKS);
+        for (int i = 0; i < tasks.length; i++) {
+            System.out.println(" " + (i + 1) + ". " + tasks[i].toString());
+        }
+        System.out.println(OUTPUT_DIVIDER);
+    }
+
+    public static void printGoodbye() {
+        System.out.println(OUTPUT_DIVIDER);
+        System.out.println(MESSAGE_GOODBYE);
+        System.out.println(OUTPUT_DIVIDER);
+    }
+
+    public static boolean isNumeric(String str) {
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    public static void markAndPrintTaskDone(Task[] tasks, int taskCount, String doneTask) {
+        String[] doneSentence = doneTask.split(" ");
+
+        // Checks if done command entered does not follow the correct format of "done {task ID}".
+        if (doneSentence.length != 2) {
+            System.out.println(OUTPUT_DIVIDER);
+            System.out.println(MESSAGE_COMMAND_WRONG_FORMAT);
+            System.out.println(MESSAGE_DONE_CORRECT_FORMAT);
+            System.out.println(OUTPUT_DIVIDER);
+            return;
+        }
+
+        // Checks if the task ID entered is numeric.
+        if (!isNumeric(doneSentence[1])) {
+            System.out.println(OUTPUT_DIVIDER);
+            System.out.println(MESSAGE_COMMAND_WRONG_FORMAT);
+            System.out.println(MESSAGE_NUMERIC_TASK_ID);
+            System.out.println(OUTPUT_DIVIDER);
+            return;
+        }
+
+        int taskToMarkDone = Integer.parseInt(doneSentence[1]);
+        // Makes sure that the task being mark done is in the task list.
+        if (taskToMarkDone > taskCount || taskToMarkDone <= 0) {
+            System.out.println(OUTPUT_DIVIDER);
+            System.out.println(MESSAGE_TASK_NOT_IN_LIST);
+            System.out.println(OUTPUT_DIVIDER);
+        } else {
+            tasks[taskToMarkDone - 1].setDone(true);
+            System.out.println(OUTPUT_DIVIDER);
+            System.out.println(MESSAGE_TASK_MARKED_DONE);
+            System.out.println("   " + tasks[taskToMarkDone - 1].toString());
+            System.out.println(OUTPUT_DIVIDER);
+        }
+    }
+
+    public static boolean canAddAndPrintTodo(Task[] tasks, int taskCount, String addedTodo) {
```

Comment:
Should the method perform 2 distinct tasks in a single function?
Is the method name appropriate for the return type 'boolean'?

Reply:
Understand why it is a boolean but it is hard to understand at a glance, maybe a change in the method name would be better as Amos pointed out